### ISSUE TITLE
Add missing aria roles

### ISSUE THREAD

**Issue by carolinan** [WordPress/twentytwenty#462]
The site header is missing the aria landmark role.

For reference: https://make.wordpress.org/themes/handbook/review/accessibility/required/#aria-landmark-roles

**Comment by pattonwebz** [comment]
There is no need to add a role='banner' to this `<header>` element as it is a direct decendent of the `<body>` tag :) See: https://developer.mozilla.org/en-US/docs/Web/Accessibility/ARIA/Roles/Banner_role

>Assistive technologies can identify the main header element of a page as the banner if is a descendant of the body element, and not nested within an article, aside, main, nav or section subsection.

**Comment by carolinan** [comment]
The explanation I got from the accessibility team is that _not all_ screen readers reads these HTML5 elements as landmarks and that this is still needed. Mind you this was some time ago. This information/statistics may be outdated. 

@afercia, your thoughts? :)

**Comment by afercia** [comment]
Hello. The problem is old screen readers (and browsers bugs) may prevent the role to be reported. This is why the accessibility team still recommends to use an explicit ARIA role in addition to the HTML5 element for `<header>`, `<nav>`, `<main>`, `<footer>`, `<aside>`.

Noting this will trigger a warning when validating via the W3C HTML validator but that's just a warning.

Latest information I can point to in this regard:

Accessible Landmarks
https://www.scottohara.me/blog/2018/03/03/landmarks.html
by Scott O'Hara
Published: Mar 3, 2018 | Updated: 2019.03.01

> At the time of writing this, there are known issues with browsers not surfacing native form elements as landmarks. Additionally, Safari and Chrome with VoiceOver on macOS High Sierra/Mojave do not recognize a footer as a contentinfo landmark, even if the footer is scoped to the body element.

> Most native HTML5 elements expose landmarks appropriately, it never hurts to test in different browser and screen reader combos.

> While it is typically not advised to use a redundant ARIA role on an element that should natively convey that role, such as `<main role="main">`, depending on your own requirements for legacy browser support, and to mitigate bugs, like the ones mentioned here, there may be times where this rule is overlooked to ensure landmarks are exposed as intended.

**Comment by pattonwebz** [comment]
Thank you @carolinan and @afercia :)

Understood, for the most accessible experience both use of the html tag for a role as well as a role being explicitly stated is best for most people.

Very good to know :)

**Comment by pattonwebz** [comment]
@nielslange I am assigning this one to you.

**Comment by nielslange** [comment]
Thanks, @pattonwebz!

**Comment by carolinan** [comment]
So we learnt that there are more landmarks that are missing their role, it also needs to be added to 
all the navs except the footer menu which already has a role.

**Comment by ianbelanger79** [comment]
We still need header `role="banner"`
